pkg/kubelet/cm/cpumanager/policy_static_test.go

```diff
@@ -1147,6 +1148,168 @@ func TestStaticPolicyAddWithResvList(t *testing.T) {
+			// remove partially used uncores from the available CPUs to simulate fully clean slate
+			stDefaultCPUSet: topoDualSocketMultiNumaPerSocketUncore.CPUDetails.CPUs().Difference(
+				cpuset.New().Union(
+					topoDualSocketMultiNumaPerSocketUncore.CPUDetails.CPUsInUncoreCaches(0),
+				).Union(
+					topoDualSocketMultiNumaPerSocketUncore.CPUDetails.CPUsInUncoreCaches(16),
+				),
+			),
+			pod: WithPodUID(
+				makeMultiContainerPod(
+					[]struct{ request, limit string }{}, // init container
+					[]struct{ request, limit string }{ // app container
+						{"2000m", "2000m"},
+						{"12000m", "12000m"},
+					},
+				),
+				"with-sidecar-and-app-container",
+			),
+		},
+		{
+			description:     "GuPodMainAndManySidecarContainer, DualSocketHTUncore, ExpectAllocOneUncore, FullUncoreAvail",
+			topo:            topoDualSocketMultiNumaPerSocketUncore,
+			numReservedCPUs: 8,
+			reserved:        cpuset.New(0, 1, 96, 97, 192, 193, 288, 289), // note 4 cpus taken from uncore 0, 4 from uncore 16
+			cpuPolicyOptions: map[string]string{
+				FullPCPUsOnlyOption:            "true",
+				PreferAlignByUnCoreCacheOption: "true",
+			},
+			stAssignments: state.ContainerCPUAssignments{},
+			// remove partially used uncores from the available CPUs to simulate fully clean slate
+			stDefaultCPUSet: topoDualSocketMultiNumaPerSocketUncore.CPUDetails.CPUs().Difference(
+				cpuset.New().Union(
+					topoDualSocketMultiNumaPerSocketUncore.CPUDetails.CPUsInUncoreCaches(0),
+				).Union(
+					topoDualSocketMultiNumaPerSocketUncore.CPUDetails.CPUsInUncoreCaches(16),
+				),
+			),
+			pod: WithPodUID(
+				makeMultiContainerPod(
+					[]struct{ request, limit string }{}, // init container
+					[]struct{ request, limit string }{ // app container
+						{"10000m", "10000m"},
+						{"2000m", "2000m"},
+						{"2000m", "2000m"},
+						{"2000m", "2000m"},
+					},
+				),
+				"with-app-container-and-multi-sidecar",
+			),
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			policy, err := NewStaticPolicy(testCase.topo, testCase.numReservedCPUs, testCase.reserved, topologymanager.NewFakeManager(), testCase.cpuPolicyOptions)
+			if err != nil {
+				t.Fatalf("NewStaticPolicy() failed with %v", err)
+			}
+
+			st := &mockState{
+				assignments:   testCase.stAssignments,
+				defaultCPUSet: testCase.stDefaultCPUSet.Difference(testCase.reserved), // ensure the cpumanager invariant
+			}
+
+			for idx := range testCase.pod.Spec.Containers {
+				container := &testCase.pod.Spec.Containers[idx]
+				err := policy.Allocate(st, testCase.pod, container)
+				if err != nil {
+					t.Fatalf("Allocate failed: pod=%q container=%q", testCase.pod.UID, container.Name)
+				}
+			}
+
+			uncoreCacheIDs, err := getPodUncoreCacheIDs(st, testCase.topo, testCase.pod)
+			if err != nil {
+				t.Fatalf("uncore cache check: %v", err.Error())
+			}
+			ids := cpuset.New(uncoreCacheIDs...)
+			if ids.Size() != 1 {
```

Comment:
Would we want to replace this check and use `expCPUAlloc` instead to show edge cases where the best effort method can allocate a single container across multiple uncores and subcontainers of a pod on multiple uncores